.github/workflows/functional-test.yaml

```diff
@@ -628,6 +627,40 @@ jobs:
           append: true
           message: |
             :x: ${{ matrix.name }} functional test cancelled. Please check [the logs](${{ env.ACTION_LINK }}) for more details
+  report-test-results:
+    name: Report test results
+    needs: [build, tests]
+    runs-on: ubuntu-latest
+    if: always()
+    env:
+      CHECKOUT_REF: ${{ needs.build.outputs.CHECKOUT_REF }}
+    steps:
+      - name: Get GitHub app token
+        uses: tibdex/github-app-token@v1
+        id: get_installation_token
+        with: 
+          app_id: ${{ env.FUNCTIONAL_TEST_APP_ID }}
+          private_key: ${{ secrets.FUNCTIONAL_TEST_APP_PRIVATE_KEY }}
+      - name: Get tests job status
+        id: get_test_status
+        run: |
+          # from: https://github.com/orgs/community/discussions/26526#discussioncomment-3252209
+          ALL_JOBS_STATUS=$(curl -X GET -s -u "admin:${{ steps.get_installation_token.outputs.token }}" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq ".jobs[] | {job_status: .conclusion}")
+          echo "All jobs status: $ALL_JOBS_STATUS"
+          TEST_STATUS="success"
+          for job_status in $(echo "$ALL_JOBS_STATUS" | jq -r '.[]'); do
+            echo "Job Status: $job_status"
+            if [[ "$job_status" == "failure" ]]; then
+              echo "Found failed test. Setting test status to failure"
+              TEST_STATUS="failure"
+              break
+            elif [[ "$job_status" == "cancelled" ]]; then
+              echo "Found cancelled test. Setting test status to cancelled"
+              TEST_STATUS="cancelled"
```

Comment:
Should we also have a break here?

Reply:
I added a break for the failure but not the cancelled because I think it's better to report "tests failed" instead of "tests cancelled" if both occur. If all tests pass but some are cancelled, we will still report the status as cancelled because we only set as "success" if we don't go in this conditional.

Reply:
ok